akka-persistence-testkit/src/main/scala/akka/persistence/testkit/internal/TestKitStorage.scala

```diff
@@ -72,58 +97,74 @@ sealed trait InMemStorage[K, R] extends InternalReprSupport[R] {
   /**
    * Sets new elements returned by updater ordered by seqnum. Sets new seqnum as max(old, max(newElemsFromUpdaterSeqNums))
    */
-  def updateOrSetNew(key: K, updater: Vector[R] => Vector[R]): Vector[R] =
-    eventsMap
-      .compute(
-        key,
-        (_: K, value: (Long, Vector[InternalRepr])) => {
-          val (sn, elems) = if (value != null) value else (0L, Vector.empty)
-          val upd = updater(elems.map(toRepr)).sortBy(reprToSeqNum)
-          (max(getLastSeqNumber(upd), sn), upd.map(toInternal))
-        })
-      ._2
-      .map(toRepr)
+  def updateOrSetNew(key: K, updater: Vector[R] => Vector[R]): Vector[R] = lock.synchronized {
```

Comment:
This function now re-implements the semantics of `.compute` which was only available on the previous collection

Reply:
I'd inline and simpify by writing it as:

```
  def updateOrSetNew(key: K, updater: Vector[R] => Vector[R]): Vector[R] = lock.synchronized {
    val (oldSn, oldElems) = eventsMap.getOrElse(key, (0L, Vector.empty))
    val newValue = {
      val upd = updater(oldElems.map(toRepr)).sortBy(reprToSeqNum)
      (max(getLastSeqNumber(upd), oldSn), upd.map(toInternal))
    }
    
    eventsMap = eventsMap.updated(key, newValue)
    newValue._2.map(toRepr)
  }
```